smartsim/_core/control/controller.py

```diff
@@ -105,6 +109,43 @@ def start(
             # it may be called seperately
             self.poll(5, True, kill_on_interrupt=kill_on_interrupt)
 
+        def _prep_launched_metadata(
+            launcher: Launcher, launched_step_name: str, step: Step
+        ) -> t.Tuple[t.Optional[str], t.Optional[str], t.Optional[bool], str, str]:
+            # NOTE: we cannot assume that the name of the launched step
+            # ``launched_step_name`` is equal to the name of the step referring
+            # to the entity ``step.name`` as is the case when an entity list is
+            # launched as a batch job
+            launched_step_map = launcher.step_mapping[launched_step_name]
+            out_file, err_file = step.get_output_files()
+            return (
+                launched_step_map.step_id,
+                launched_step_map.task_id,
+                launched_step_map.managed,
+                out_file,
+                err_file,
+            )
+
+        # FIXME: These list comprehensions are unreadable
```

Comment:
I can _read_ them!

Reply:
LOL!! Moved mapping of `LaunchedManifest` data to a separate (hopefully more readable) helper method